src/featuretransforms.jl

```diff
@@ -0,0 +1,154 @@
+
+The transform can be applied to a subselection of components via a [`Pattern`](@ref) `key`.
+Otherwise, components are selected by the desired `dims`.
+
+Keyword arguments including `dims` are passed to the appropriate `FeatureTransforms` method
+for a component.
+"""
+function FeatureTransforms.apply!(
+    ds::KeyedDataset, t::Transform, keys...;
+    dims=:, kwargs...
+)
+    return map(ds, _transform_pattern(keys, dims)...) do a
+        FeatureTransforms.apply!(a, t; dims=dims, kwargs...)
+    end
+end
+
+"""
+    FeatureTransforms.apply_append(
+        ds::KeyedDataset, t::Transform, [key];
+        dims=:, inner=false, component_name=:component, kwargs...
+    )
+
+Apply the `Transform` to each component of the [`KeyedDataset`](@ref).
+
+The transform can be applied to a subselection of components via a [`Pattern`](@ref) `key`.
+Otherwise, components are selected by the desired `dims`.
+
+If `inner=true`, perform `FeatureTransforms.apply_append` on each component,
+returning a new dataset with the same constraints, but transformed components.
+
+Otherwise, transform each component using `FeatureTransforms.apply`, and append
+to a copy of the dataset as a new component called `component_name`.
+
+Keyword arguments including `dims` are passed to the appropriate `FeatureTransforms` method
+for a component.
+
+# Example
+```jldoctest
+julia> using AxisKeys, FeatureTransforms; using AxisSets: KeyedDataset, Pattern, flatten;
+
+julia> ds = KeyedDataset(
+           flatten([
+               :train => [
+                   :load => KeyedArray([7.0 7.7; 8.0 8.2; 9.0 9.9]; time=1:3, loc=[:x, :y]),
+                   :price => KeyedArray([-2.0 4.0; 3.0 2.0; -1.0 -1.0]; time=1:3, id=[:a, :b]),
+               ],
+               :predict => [
+                   :load => KeyedArray([7.0 7.7; 8.1 7.9; 9.0 9.9]; time=1:3, loc=[:x, :y]),
+                   :price => KeyedArray([0.5 -1.0; -5.0 -2.0; 0.0 1.0]; time=1:3, id=[:a, :b]),
+               ]
+           ])...
+       );
+
+julia> p = Power(2);
+
+julia> r = FeatureTransforms.apply_append(ds, p, (:_, :price, :_); component_name=:price2);
+
+julia> [k => parent(parent(v)) for (k, v) in r.data]
+6-element Vector{Pair{Tuple{Symbol, Symbol}, Matrix{Float64}}}:
+     (:train, :load) => [7.0 7.7; 8.0 8.2; 9.0 9.9]
+    (:train, :price) => [-2.0 4.0; 3.0 2.0; -1.0 -1.0]
+   (:predict, :load) => [7.0 7.7; 8.1 7.9; 9.0 9.9]
+  (:predict, :price) => [0.5 -1.0; -5.0 -2.0; 0.0 1.0]
+   (:train, :price2) => [4.0 16.0; 9.0 4.0; 1.0 1.0]
+ (:predict, :price2) => [0.25 1.0; 25.0 4.0; 0.0 1.0]
+```
+"""
+function FeatureTransforms.apply_append(
+    ds::KeyedDataset, t::Transform, keys...;
+    dims=:, inner=false, component_name=:component, kwargs...
+)
+    patterns = _transform_pattern(keys, dims)
+
+    if inner  # batched apply_append on each component
+        return map(ds, patterns...) do a
+            FeatureTransforms.apply_append(a, t; dims=dims, kwargs...)
+        end
```

Comment:
Just thought this could be handy and worth including.

Reply:
I could see this being useful, but I'm not sure we have enough use-cases yet. That being said, it wouldn't be too hard to deprecated if it isn't useful.

Reply:
I'm not sure how useful this is right away either, and it might complicate the code and tests if we have to support both `batch=true/false`  .

given the implementation is rather easy (just a `map` over the components) it should be straightforward for users to do it themselves and hold off doing it here until we know it's worth doing.